### ISSUE TITLE
Can't render content plugins added to context

### ISSUE THREAD

**Issue by alsoicode** [jnordberg/wintersmith#277]
Hello. I'm having some difficulties getting a custom plugin, modeled after the paginator plugin, to work. The plugin code:

    module.exports = (env, callback) ->

        getWorkSamples = (contents) ->
            workSamples = contents['work']._.directories.map (item) -> item.index
            # sort work samples by order
            workSamples.sort (a, b) -> a.metadata.order - b.metadata.order

            samples = []
            for workSample in workSamples
                samples.push new WorkSample workSample.filepath, workSample.metadata

            return samples

        class WorkSample extends env.ContentPlugin

            constructor: (filepath, metadata) ->
                @filepath = filepath
                @title = metadata.title
                @thumbnailUrl = metadata.thumbnailUrl

            getFilename: ->
                @filepath.relative

            getUrl: ->
                ''

            getView: -> (env, locals, contents, templates, callback) ->
                template = templates['work/detail.html']
                ctx = {@workSample}
                env.utils.extend ctx, locals
                template.render ctx, callback

        env.registerGenerator 'work', (contents, callback) ->
            samples = getWorkSamples contents
            rv = {workSamples: samples}
            callback null, rv

        env.helpers.getWorkSamples = getWorkSamples

        callback()

If I `console.log rv` in the `env.registerGenerator` function, I can see the array of objects, however, when I attempt to render them via [wintersmith-nunjucks](https://github.com/jbuck/wintersmith-nunjucks):

    {% for workSample in workSamples %}
        {{ workSample.title }}
    {% endfor %}

There is no output at all. What am I doing wrong?

**Comment by alsoicode** [comment]
Thinking I might be missing a property or method, I also tried changing the `WorkSample` to inherit from `env.plugins.Page` to no avail. This is the last piece of my site I need to get working with Wintersmith. I'm not sure how this is so different from pages and the articles shown on the index.jade template...

**Comment by jnordberg** [comment]
Hmm, not sure exactly what you are trying to do but it looks like you're never assigning `@workSample` anywhere. Maybe you meant `ctx = {workSamples: contents.workSamples}` in your view?

**Comment by alsoicode** [comment]
Thanks for the reply. I'm trying to generate this page on my site: https://brandonftaylor.com/work/ which is currently powered by Django. I enjoy working with Node, so I thought I'd give Wintersmith a try.

I've created a folder structure that matches the "articles" from the example, and I'm able to get all of the work samples from the directory structure and metadata of the index.md files contained in each.

I wasn't sure if I should create a ContentPlugin, TemplatePlugin or a Generator, so I just attempted to replicate the "paginator" plugin substituting my "WorkSample" plugin for the "Page" with its nested Articles.

I guess I'm confused about where to assign the values I get back from `getWorkSamples` so that they can be iterated over in a top-level page.

Is the `getView` function on the `WorkSample` class not for an instance? I was thinking that was the function that would be called to render an instance.

Thank you very much for your time.
